scope/test.h

```diff
@@ -76,8 +80,10 @@ namespace scope {
       if (*msg) {
         buf << msg << " ";
       }
-      buf << "Expected[" << (mis.first - ebeg) << "]: " << *mis.first
-          << ", Actual[" << (mis.second - abeg) << "]: " << *mis.second;
+      buf << "Expected["
+          << std::distance(ebeg, mis.first) << "]: " << *mis.first
+          << ", Actual["
+          << std::distance(abeg, mis.second) << "]: " << *mis.second;
```

Comment:
And then the message should read:

Expected at index X: *mis.first, Actual: *mis.second. Sizes of expected and actual are Y, Z.

This way if there is a size mismatch, you get more information—the position where the mismatch occurs and the value that's there. Previously we just reported on the size mismatch.